### PR TITLE
Added performance tip for parentheses in expressions to docs

### PR DESCRIPTION
**Pull request by goelakash**
#14821 Please review.

### CODE REVIEW

**Comment by nalimilan** [comment]
Thanks! However, I think that if add this we also need to explain why this happens and whether something is going to be done about this in future releases. Unfortunately, I can't tell about these two points myself.

**Comment by goelakash** [comment]
@nalimilan @jiahao Could you tell me anything else that I can add (such as the reasons for the performance degradation) so that I may finish this PR?

**Comment by nalimilan** [comment]
@JeffBezanson @vtjnash Do you think that effect of parentheses on performance is likely to be fixed soon? Could you explain why it happens in a few words?

**Comment by yuyichao** [comment]
Splatting penalty.

File: doc/manual/performance-tips.rst [hunk 1]
```diff
@@ -800,6 +800,49 @@ Taken to its extreme, pre-allocation can make your code uglier, so
 performance measurements and some judgment may be required.
 
 
+Use parentheses in arithmetic operations
+----------------------------------------
+
+If your code has a long arithmetic operation involving `+` or `*` operators,
+then consider using parentheses to chain upto four or five operations together.
```
Comment (tkelman) [inline doc/manual/performance-tips.rst]: "up to" is two words, not one

File: doc/manual/performance-tips.rst [hunk 2]
```diff
@@ -800,6 +800,51 @@ Taken to its extreme, pre-allocation can make your code uglier, so
 performance measurements and some judgment may be required.
 
 
+Use parentheses in arithmetic operations
+----------------------------------------
+
+If your code has a long arithmetic operation involving `+` or `*` operators,
```
Comment (tkelman) [inline doc/manual/performance-tips.rst]: double backticks in rst

File: doc/manual/performance-tips.rst [hunk 3]
```diff
@@ -800,6 +800,51 @@ Taken to its extreme, pre-allocation can make your code uglier, so
 performance measurements and some judgment may be required.
 
 
+Use parentheses in arithmetic operations
```
Comment (nalimilan) [inline doc/manual/performance-tips.rst]: I think this should be "in long arithmetic expressions".

**Comment by jiahao** [comment]
I think we can accept this and change it later if and when the performance tip is no longer necessary. Thanks

File: doc/manual/performance-tips.rst [hunk 4]
```diff
@@ -800,6 +800,51 @@ Taken to its extreme, pre-allocation can make your code uglier, so
 performance measurements and some judgment may be required.
 
 
+Use parentheses in long arithmetic operations
+---------------------------------------------
+
+If your code has a long arithmetic operation involving ``+`` or ``*`` operators,
+then consider using parentheses to chain up to four or five operations together.
+This is to avoid the splatting penalty from a longer list of arguments to function
+return statement.
```
Comment (tkelman) [inline doc/manual/performance-tips.rst]: there's nothing specific to `return` about this

Comment (jiahao) [inline doc/manual/performance-tips.rst]: I read "return" as the output value of the expression, but I can fix it

**Comment by tkelman** [comment]
Note also the rst errors from `make docs`:
```
/home/Tony/julia32/doc/manual/performance-tips.rst:962: ERROR: Unexpected indentation.
/home/Tony/julia32/doc/manual/performance-tips.rst:963: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/Tony/julia32/doc/manual/performance-tips.rst:968: ERROR: Unexpected indentation.
/home/Tony/julia32/doc/manual/performance-tips.rst:969: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/Tony/julia32/doc/manual/performance-tips.rst:971: WARNING: Definition list ends without a blank line; unexpected unindent.
/home/Tony/julia32/doc/manual/performance-tips.rst:980: ERROR: Unexpected indentation.
/home/Tony/julia32/doc/manual/performance-tips.rst:981: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/Tony/julia32/doc/manual/performance-tips.rst:986: ERROR: Unexpected indentation.
/home/Tony/julia32/doc/manual/performance-tips.rst:987: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/Tony/julia32/doc/manual/performance-tips.rst:989: WARNING: Definition list ends without a blank line; unexpected unindent.
```

**Pull request merged**